pkg/server/ca/manager.go

```diff
@@ -811,6 +818,40 @@ func (m *Manager) notify(ctx context.Context, event string, advise bool, pre fun
 	return nil
 }
 
+func (m *Manager) filterInvalidEntries(ctx context.Context, entries *journal.Entries) ([]*JWTKeyEntry, []*X509CAEntry, error) {
+	bundle, err := m.fetchOptionalBundle(ctx)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	filteredEntriesJwtKeys := []*JWTKeyEntry{}
+	filteredEntriesX509CAs := []*X509CAEntry{}
+
+	// certificate append to the database whenever upstreamClient does not exist
+	if m.upstreamClient != nil {
+		filteredEntriesX509CAs = entries.X509CAs
+	} else {
+		for _, entry := range entries.GetX509CAs() {
+			if containsX509CA(bundle.RootCas, entry.Certificate) {
+				filteredEntriesX509CAs = append(filteredEntriesX509CAs, entry)
+			} else {
+				m.c.Log.Warn("Dropping X509 CA as it not found in bundle")
+			}
+		}
```

Comment:
Similar comment here about avoiding if/else for readability

```suggestion
		for _, entry := range entries.GetX509CAs() {
			if containsX509CA(bundle.RootCas, entry.Certificate) {
				filteredEntriesX509CAs = append(filteredEntriesX509CAs, entry)
				continue
			}

			m.c.Log.Warn("Dropping X509 CA as it not found in bundle")
		}
```